numpy/core/src/multiarray/einsum.c.src

```diff
@@ -2225,7 +2225,7 @@ get_combined_dims_view(PyArrayObject *op, int iop, char *labels)
         }
         else {
             /* Update the combined axis dimensions and strides */
-            i = idim + combineoffset;
+            int i = icombinemap[idim + combineoffset];
             if (combineoffset < 0 && new_dims[i] != 0 &&
```

Comment:
Should this `combineoffset` check come first?

Reply:
Not really. `i` has to be mapped to the value of `icombine` (which is only advanced when a label is seen for the first time) of the first time this label was found, which was in dimension `idim + combineoffset`, and this has to be done regardless of whether this is the first time we see this label or not.

Reply:
My thinking is that this could index out of range if `combineoffset` is negative